analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantDeclaration.CSharp12.ArraySize.Fixed.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace AliasAnyType
```

Comment:
I cannot see tests for collection expressions, I think we should add them, if we look at the title of the rule "Declarations and initializations should be as concise as possible" some of the compliant examples from previous C# versions are now FNs.

```cs
int[] ints = [1, 2, 3]; // Compliant
ints = new[] { 1, 2, 3, 4 }; // FN can be written ints = [1, 2, 3, 4]
ints = new int[] { }; // FN can be written ints = []
[...] etc
```

Reply:
Indeed! In the Google document, I have specified `Implement`, but I have not added whether it's an FN or not, because, while I could try to make that determination, I think the rule requires some general rework. 

Moreover, it seems there may be some overlapping with IDE0300 (Collection initialization can be simplified), which we may not want to re-implement. So maybe the rule is good as it is, and we rely on VS code fixes to cover collection expressions.

Anyway, I have added the tests you mentioned above, and a few other more, to document the current behavior and marked those cases as FN, creating a [GitHub issue for the FN](https://github.com/SonarSource/sonar-dotnet/issues/8115). In the GitHub issue, I have mentioned that the FN may need to be evaluated and possibly challenged.